src/test/java/cs2103/v15_1j/jimjim/controller/JJControllerTest.java

```diff
@@ -123,6 +124,15 @@ public void testAdd() {
     public void testDelete() {
         controller.execute("Prepare for German exams");
         controller.execute("Buy flowers for her by Tuesday");
+        
+        for(DeadlineTask t: controller.getMasterList().getDeadlineTasksList()){
+        	controller.states.displayList.add(t);
+        }
+        
+        for(FloatingTask t: controller.getMasterList().getFloatingTasksList()){
+        	controller.states.displayList.add(t);
+        }
+        
```

Comment:
Reading this test make me realize this is not a good design. The logic is supposed to work independently of the UI, i.e. I can write a text UI for it instead of a GUI, and all the text UI gonna do is to feed the controller with user's command. However, it doesn't work in this design. Part of the controller relies on the GUI to work (thus you have to insert those 2 for loops to imitate what the UI does). The controller is now dependent on the UI.